File: src/types/elements/createLearningAssetDto.ts
import { CreateMetaDto } from "./createMetaDto";
import { LearningAssetType } from "../../api/elements/dto/anyElementDto";

export interface CreateLearningAssetDto {
	id: string;
	meta: CreateMetaDto;
	type: LearningAssetType;
	splits: string[];
	initialPriorityRank?: number;
	pdfBytesBase64?: string;
	pdfPageCount?: number;
}

File: src-tauri/src/elements/dto/create_learning_asset_dto.rs
use serde::Deserialize;
use uuid::Uuid;

use super::create_meta_dto::CreateMetaDto;
use crate::elements::entities::learning_asset::LearningAssetType;

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateLearningAssetDto {
    pub id: Uuid,
    pub meta: CreateMetaDto,
    pub r#type: LearningAssetType,
    pub splits: Vec<String>,
    pub initial_priority_rank: Option<i64>,
    #[serde(default)]
    pub pdf_bytes_base64: Option<String>,
    #[serde(default)]
    pub pdf_page_count: Option<u32>,
}
